src/MainWindow.h

```diff
@@ -0,0 +1,257 @@
+     * @return The original string but with spaces removed.
+     */
+    QString removeSpaces(QString p_stringWithSpaces);
+
+    /**
+     * @brief Clears the name, amount due, and due date fields in preparation for allowing a user to enter another bill.
+     */
+    void clearBillWidget();
+
+    /**
+     * @brief Organizes various buttons used for interacting with the bill table widget into a layout.
+     */
+    void createButtonGridLayout();
+
+    const QString m_APP_NAME = "PersonalFinanceTool"; //!< The name of the application.
+    const QString m_CONFIG_FILE_NAME = m_APP_NAME + ".ini"; //!< The name of the config file.
+    const QString m_CONFIG_PARENT_FOLDER = "config/"; //!< The parent folder of the config file.
+    const QString m_CONFIG_FILE_DIRECTORY = m_CONFIG_PARENT_FOLDER + m_CONFIG_FILE_NAME; //!< The path where the config file should be read/generated if absent.
+    const QString m_CONFIG_FILE_GENERATE_FAIL_BOX_PRIMARY_TEXT = "Failure To Generate Configuration File"; //!< The title of the config file generation failure message box.
+    const QString m_CONFIG_FILE_GENERATE_FAIL_BOX_INFO_TEXT = "The configuration file could not be generated in " + m_CONFIG_PARENT_FOLDER; //!< The informative text of the config file generation failure message box.
+    const QString m_CONFIG_CORRUPT_FILE_BOX_PRIMARY_TEXT = "Configuration File Is Corrupt"; //!< The title of the config file corrupt message box.
+    const QString m_CONFIG_CORRUPT_FILE_BOX_INFO_TEXT = "The configuration file could not be opened, a new one will be created following the welcoming sequence."; //!< The informative text of the corrupt config file message box.
+    const QString m_WELCOME_BOX_PRIMARY_TEXT = "Welcome to the Personal Finance Tool!"; //!< The welcome message box title.
+    const QString m_WELCOME_BOX_INFO_TEXT = "On the subsequent screens, you will be asked to provide some financial information."; //!< The welcome message box informative text.
+    const QString m_TOTAL_AMOUNT_AVAILABLE_STRING = "Total Amount Available: $"; //!< The total amount available represented as a string.
+    const QString m_ASK_FOR_AMOUNT_AVAILABLE_TEXT = "Please enter the total amount of money available in USD."; //!< The total amount available informative text.
+    const QString m_INVALID_KEY_BOX_PRIMARY_TEXT = "Invalid Configuration File Key/Value Pair"; //!< The invalid key message box title.
+    const QString m_INVALID_KEY_BOX_INFO_TEXT = "A key/value pair from " + m_CONFIG_FILE_NAME + " is invalid. Please correct the file to have a single value per key and relaunch the application."; //!< The invalid key message box informative text.
+    const QString m_MISSING_BILL_DETAILS_BOX_PRIMARY_TEXT = "Missing Bill Information"; //!< The missing bill details box title.
+    const QString m_MISSING_BILL_DETAILS_BOX_INFO_TEXT = "Please enter all bill information."; //!< The informative text of the missing bill details message box.
```

Comment:
Make this more specific. Perhaps *Please enter a name, amount due, and due date for each bill.*

Reply:
Made more specific.